qutebrowser/utils/utils.py

```diff
@@ -773,3 +773,31 @@ def libgl_workaround() -> None:
     libgl = ctypes.util.find_library("GL")
     if libgl is not None:  # pragma: no branch
         ctypes.CDLL(libgl, mode=ctypes.RTLD_GLOBAL)
+
+
+def parse_duration(duration: str) -> int:
+    """Parse duration in format XhYmZs into milliseconds duration."""
+    if duration.isdigit():
+        # For backward compatibility return milliseconds
+        return int(duration)
+
+    match = re.search(
+        r'^(?P<hours>[0-9]+(\.[0-9])*h)?\s*'
+        r'(?P<minutes>[0-9]+(\.[0-9])*m)?\s*'
+        r'(?P<seconds>[0-9]+(\.[0-9])*s)?$',
```

Comment:
This allows something like `1.1.1s` as well - it happens to not crash because `float()` raises `ValueError` which we already handle, but I'd still prefer to handle it correctly in the regex here. Wouldn't something like:

```suggestion
        r'^(?P<hours>[0-9]+(\.[0-9]+)?h)?\s*'
        r'(?P<minutes>[0-9]+(\.[0-9]+)?m)?\s*'
        r'(?P<seconds>[0-9]+(\.[0-9]+)?s)?$',
```

be more correct?